### ISSUE TITLE
Sporadic exception with editor

### ISSUE THREAD

**Issue by vepanimas** [jimkyndemeyer/js-graphql-intellij-plugin#387]
Reproduced with multiple projects opened at the same time. 

WebStorm 2020.3 Snapshot
Build #WS-203.3498, built on September 10, 2020

```
java.lang.Throwable: Light files should have PSI only in one project, existing=com.intellij.psi.SingleRootFileViewProvider{vFile=LightVirtualFile: /a.dummy, content=VirtualFileContent{size=0}, eventSystemEnabled=true} in Project(name=tour-of-heroes-v2, containerState=COMPONENT_CREATED, componentStore=/Users/kristina.Pchelintseva/WebstormProjects/myProj/tour-of-heroes-v2), requested in Project(name=untitled55, containerState=COMPONENT_CREATED, componentStore=/Users/kristina.Pchelintseva/WebstormProjects/trash/untitled55)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:159)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.checkHasNoOtherPsi(FileManagerImpl.java:205)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.findViewProvider(FileManagerImpl.java:194)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.findFile(FileManagerImpl.java:368)
	at com.intellij.psi.impl.PsiDocumentManagerBase.getPsiFile(PsiDocumentManagerBase.java:142)
	at com.intellij.psi.impl.PsiDocumentManagerBase.getPsiFile(PsiDocumentManagerBase.java:93)
	at com.intellij.psi.impl.PsiDocumentManagerImpl.getPsiFile(PsiDocumentManagerImpl.java:64)
	at com.intellij.lang.jsgraphql.v1.ide.editor.JSGraphQLQueryContextCaretListener$1.caretPositionChanged(JSGraphQLQueryContextCaretListener.java:39)
	at jdk.internal.reflect.GeneratedMethodAccessor65.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:123)
	at com.intellij.util.EventDispatcher.lambda$createMulticaster$1(EventDispatcher.java:86)
	at com.sun.proxy.$Proxy93.caretPositionChanged(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor65.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:123)
	at com.intellij.util.EventDispatcher.lambda$createMulticaster$1(EventDispatcher.java:86)
	at com.sun.proxy.$Proxy93.caretPositionChanged(Unknown Source)
	at com.intellij.openapi.editor.impl.CaretModelImpl.fireCaretPositionChanged(CaretModelImpl.java:545)
	at com.intellij.openapi.editor.impl.CaretImpl.lambda$moveToOffset$0(CaretImpl.java:128)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:421)
	at com.intellij.openapi.editor.impl.CaretImpl.moveToOffset(CaretImpl.java:105)
	at com.intellij.openapi.editor.CaretModel.moveToOffset(CaretModel.java:91)
	at com.intellij.openapi.editor.CaretModel.moveToOffset(CaretModel.java:77)
	at com.intellij.ui.EditorTextField.createEditor(EditorTextField.java:546)
	at com.intellij.ui.ComboboxEditorTextField.createEditor(ComboboxEditorTextField.java:45)
	at com.intellij.ui.EditorComboBoxEditor$1.createEditor(EditorComboBoxEditor.java:42)
	at com.intellij.ui.EditorTextField.initEditor(EditorTextField.java:413)
	at com.intellij.ui.EditorTextField.addNotify(EditorTextField.java:398)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at com.intellij.openapi.ui.ComboBox.addNotify(ComboBox.java:199)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/javax.swing.JComponent.addNotify(JComponent.java:4791)
	at java.desktop/javax.swing.JRootPane.addNotify(JRootPane.java:733)
	at java.desktop/java.awt.Container.addNotify(Container.java:2800)
	at java.desktop/java.awt.Window.addNotify(Window.java:786)
	at java.desktop/java.awt.Dialog.addNotify(Dialog.java:752)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.addNotify(DialogWrapperPeerImpl.java:639)
	at java.desktop/java.awt.Window.pack(Window.java:824)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:650)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:438)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1692)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1651)
	at com.intellij.refactoring.ui.RefactoringDialog.show(RefactoringDialog.java:103)
	at com.intellij.refactoring.rename.PsiElementRenameHandler.rename(PsiElementRenameHandler.java:201)
	at com.intellij.refactoring.rename.PsiElementRenameHandler.rename(PsiElementRenameHandler.java:167)
	at com.intellij.refactoring.rename.PsiElementRenameHandler.rename(PsiElementRenameHandler.java:162)
	at com.intellij.refactoring.rename.PsiElementRenameHandler.invoke(PsiElementRenameHandler.java:114)
	at com.intellij.refactoring.rename.PsiElementRenameHandler.invoke(PsiElementRenameHandler.java:66)
	at com.intellij.refactoring.rename.inplace.VariableInplaceRenameHandler.performDialogRename(VariableInplaceRenameHandler.java:133)
	at com.intellij.refactoring.rename.inplace.MemberInplaceRenameHandler$1.pass(MemberInplaceRenameHandler.java:73)
	at com.intellij.refactoring.rename.inplace.MemberInplaceRenameHandler$1.pass(MemberInplaceRenameHandler.java:67)
	at com.intellij.lang.javascript.refactoring.JSDefaultRenameProcessor.substituteElementToRename(JSDefaultRenameProcessor.java:360)
	at com.intellij.refactoring.rename.inplace.MemberInplaceRenameHandler.doRename(MemberInplaceRenameHandler.java:67)
	at com.intellij.refactoring.rename.inplace.VariableInplaceRenameHandler.invoke(VariableInplaceRenameHandler.java:77)
	at com.intellij.refactoring.actions.BaseRefactoringAction.performRefactoringAction(BaseRefactoringAction.java:156)
	at com.intellij.refactoring.rename.RenameHandler2Renamer.performRename(RenameHandler2Renamer.java:35)
	at com.intellij.refactoring.actions.RenameElementAction.actionPerformed(RenameElementAction.java:67)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:281)
	at com.intellij.refactoring.rename.inplace.TemplateInlayUtil.doRename(TemplateInlayUtil.kt:224)
	at com.intellij.refactoring.rename.inplace.TemplateInlayUtil.access$doRename(TemplateInlayUtil.kt:42)
	at com.intellij.refactoring.rename.inplace.TemplateInlayUtil$renamePanel$1.consume(TemplateInlayUtil.kt:211)
	at com.intellij.refactoring.rename.inplace.TemplateInlayUtil$renamePanel$1.consume(TemplateInlayUtil.kt:42)
	at com.intellij.openapi.project.DumbAwareAction$SimpleDumbAwareAction.actionPerformed(DumbAwareAction.java:80)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:281)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:611)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$3(IdeKeyEventDispatcher.java:671)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:95)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:670)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:621)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:518)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:472)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:221)
	at com.intellij.ide.IdeEventQueue.dispatchKeyEvent(IdeEventQueue.java:892)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:837)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:454)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:773)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:453)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:823)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:507)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by olivernybroe** [comment]
This exception also happens for me several times a day in phpstorm 2020.3 EAP

**Comment by vepanimas** [comment]
Fixed in #400
